chain/fee_manager_test.go

```diff
@@ -0,0 +1,194 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	//"fmt"
+	"time"
+	"testing"
+
+	"github.com/ava-labs/hypersdk/math"
+
+	"go.uber.org/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnitPrice(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Mock VM
+	rules := NewMockRules(ctrl)
+	rules.EXPECT().GetMinUnitPrice().Return(Dimensions{100, 100, 100, 100, 100})
+	rules.EXPECT().GetMaxBlockUnits().Return(Dimensions{1_800_000, 2_000, 2_000, 2_000, 2_000})
+
+	// Similar to vm.go L326
+	minUnitPrice := rules.GetMinUnitPrice()
+
+	var d Dimensions
+	feeManager := NewFeeManager(nil)
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		// Set unit prices for different dimensions
+		feeManager.SetUnitPrice(i, minUnitPrice[i])
+		d[i] = uint64(i)
+	}
+
+	// Check unit prices
+	unitPrices := feeManager.UnitPrices()
+	for i := Dimension(0); i < FeeDimensions; i++ {
+		require.Equal(unitPrices[i], feeManager.UnitPrice(i))
+	}
+
+	// Similar to spam.go
```

Comment:
Would recommend explaining what you are testing here rather than just saying it is similar to other usage.